test.py

```diff
@@ -382,6 +382,33 @@ def test_restore(self):
         with open(join(path2, 'foobar')) as f:
             self.assertEqual(f.read(), 'foobar')
 
+    def test_clone(self):
+        """ Check we can clone as a new volume
+        """
+        # create a volume with a file
+        name = PREFIX_TEST_VOLUME + uuid.uuid4().hex
+        path = join(VOLUMES_PATH, name)
+        self.create_a_volume_with_a_file(name)
+
+        # clone a different volume
+        name2 = PREFIX_TEST_VOLUME + uuid.uuid4().hex
+        path2 = join(VOLUMES_PATH, name2)
+
+        # clone name as new volume name2
+        resp = self.app.post('/VolumeDriver.Clone',
+                             json.dumps({'Name': name,
+                                         'Target': name2}))
+
+        # modify the file in name2 (new volume)
+        with open(join(path2, 'foobar'), 'w') as f:
+            f.write('modified2 foobar')
```

Comment:
Before updating the file in the freshly cloned volume could you make sure it contains foobar to make sure we clone properly create a snapshot? 
by adding following snippet before this part:

``` 
+        # check the cloned volume has a copy of the original content
+        with open(join(path2, 'foobar')) as f:
+            self.assertEqual(f.read(), 'foobar')
```

I don't remember if we snapshot twice in an existing volume, but this is an other story related to the snapshot method don't worry about it here!